atuin-client/src/import/fish.rs

```diff
@@ -0,0 +1,209 @@
+// import old shell history!
+// automatically hoover up all that we can find
+
+use std::{
+    fs::File,
+    io::{self, BufRead, BufReader, Read, Seek},
+    path::{Path, PathBuf},
+};
+
+use chrono::prelude::*;
+use chrono::Utc;
+use directories::UserDirs;
+use eyre::{eyre, Result};
+
+use super::{count_lines, Importer};
+use crate::history::History;
+
+#[derive(Debug)]
+pub struct Fish<R> {
+    file: BufReader<R>,
+    strbuf: String,
+    loc: usize,
+}
+
+impl<R: Read + Seek> Fish<R> {
+    fn new(r: R) -> Result<Self> {
+        let mut buf = BufReader::new(r);
+        let loc = count_lines(&mut buf)?;
+
+        Ok(Self {
+            file: buf,
+            strbuf: String::new(),
+            loc,
+        })
+    }
+}
+
+impl<R: Read> Fish<R> {
+    fn new_entry(&mut self) -> io::Result<bool> {
+        let inner = self.file.fill_buf()?;
+        Ok(inner.starts_with(b"- "))
+    }
+}
+
+impl Importer for Fish<File> {
+    const NAME: &'static str = "fish";
+
+    fn histpath() -> Result<PathBuf> {
+        let user_dirs = UserDirs::new().unwrap();
+        let home_dir = user_dirs.home_dir();
+
+        let histpath = home_dir.join(".local/share/fish/fish_history");
```

Comment:
There did used to be an alternative path https://github.com/fish-shell/fish-shell/issues/862#issuecomment-19138822 but that's very old (at least pre 2013). It might also have some parsing quirks, so I'm going to forget it exists.

It's worth checking to find out if this is the same on macos too.

Reply:
https://fishshell.com/docs/current/interactive.html#searchable-command-history